DigitalLearningSolutions.Web/Controllers/Register/RegisterDelegateByCentreController.cs

```diff
@@ -202,7 +207,41 @@ public IActionResult Summary()
         [HttpPost]
         public IActionResult Summary(SummaryViewModel model)
         {
-            return new OkResult();
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var candidateNumber =
+                registrationService.RegisterDelegateByCentre(
+                    RegistrationMappingHelper.MapToDelegateRegistrationModel(data)
+                );
+
+            switch (candidateNumber)
+            {
+                case "-1":
+                    return RedirectToAction("Error", "LearningSolutions");
+                case "-4":
+                    return RedirectToAction("Index");
+            }
+
+            TempData.Clear();
+            TempData.Add("delegateNumber", candidateNumber);
+            TempData.Add("emailSent", data.ShouldSendEmail);
+            TempData.Add("passwordSet", data.IsPasswordSet);
+            return RedirectToAction("Confirmation");

```

Comment:
Is there a particular reason we don't just perform whatever is done in the Confirmation action here instead? I'm just thinking it would save having to worry about TempData as much.

Reply:
Oh yeah that would simplify things.

Reply:
But then it would have the Summary URL wouldn't it.

Reply:
Oh yeah that's true. Seeing as how this method only redirects to the confirmation page or some other pages in the case of an error, you might be able to get away with changing the Route to be Confirmation (might also need to make the Summary page form specify the action name). Although I think we have some other pages that use this TempData -> Redirect methodology, so this might be fine as is. I'll approve the PR anyway, and you can decide what you think would be better.